packages/slate-editor/src/extensions/mentions/components/MentionsDropdown.tsx

```diff
@@ -37,36 +52,50 @@ export const MentionsDropdown = <V extends object>({
         ensureChildInView(suggestionsPanelRef.current, index, AUTO_SCROLL_SAFETY_MARGIN);
     }, [index, suggestionsPanelRef]);
 
-    useEffect(() => {
-        if (target && options.length > 0 && suggestionsPanelRef.current) {
-            const domRange = EditorCommands.toDomRange(editor, target);
-
-            if (!domRange) {
-                return;
-            }
-
-            const rect = RangeFix.getBoundingClientRect(domRange);
-
-            if (!rect) {
-                return;
-            }
-
-            const { left, height, top } = rect;
-            suggestionsPanelRef.current.style.top = `${top + window.pageYOffset + height}px`;
-            suggestionsPanelRef.current.style.left = `${left + window.pageXOffset}px`;
-        }
-    }, [editor, index, options.length, target]);
+    const referenceElement = useMemo(() => {
+        return {
+            getBoundingClientRect(): ClientRect {
+                if (!target) {
+                    return EMPTY_RECT;
+                }
+                const domRange = EditorCommands.toDomRange(editor, target);
+                if (!domRange) {
+                    return EMPTY_RECT;
+                }
+                const rect = RangeFix.getBoundingClientRect(domRange);
+                if (!rect) {
+                    return EMPTY_RECT;
+                }
+                const { left, right, top, bottom, width, height } = rect;
+                return {
+                    left,
```

Comment:
Wouldn't a `...rest` spread be shorter and more suitable here?

Reply:
Yeah, good point. This is not obvious, but we cannot use spread here. I've added a comment about it: 78268b68a09b5fd88a2ea6a9fc9be01f47b3ae38

Reply:
Oh interesting, thanks!